packages/lib/src/io-command.ts

```diff
@@ -618,14 +649,22 @@ export abstract class SelectingInputOutputAPICommand<I, O, L> extends SelectingI
 export abstract class SelectingOutputAPICommandBase<ID, O, L> extends APICommand {
 	protected abstract async getIdFromUser(items: L[]): Promise<ID>
 
-	protected async processNormally(id: ID | undefined,
+	protected abstract async translateToId(idOrIndex: ID | string,
+		listFunction: ListCallback<L>): Promise<ID>
+
+	protected acceptIndexId = false
+
+	protected async processNormally(idOrIndex: ID | string | undefined,
 			listCallback: ListCallback<L>,
 			actionCallback: GetCallback<ID, O>): Promise<void> {
 		try {
 			let inputId: ID
-			if (id) {
-				this.log(`using id from command line = ${id}`)
-				inputId = id
+			if (idOrIndex) {
+				if (this.acceptIndexId || typeof idOrIndex !== 'string' || !idOrIndex.match(validIndex)) {
```

Comment:
I don't understand the why we're using `typeof` for the middle expression (`|| typeof idOrIndex !== 'string'`). After staring at it for a few minutes, it seems like the purpose of this clause is to pass when no `idOrIndex` is passed but that would be more clear IMO with a simple `|| !idOrIndex`.

Reply:
Without the `typeof idOrIndex !== 'string'` typescript complains about the `idOrIndex.match()` call because `idOrIndex` can either be a string or an object of the `ID` class, and `match()` doesn't accept the latter

Reply:
And it's not a case of no idOrIndex being passed. The outer statement has already tested `if (idOrIndex)`. The idea is to fail when `this.acceptIndexId` is false and `idOrIndex` is an index, so that we get a descriptive error message rather than just the 403 that results from passing an index to the API call

Reply:
Oh, I hadn't thought of the type checking for some reason.